Discovery configuration to support geospatial search

kshepherd opened the pull request:
## References
Required for complete testing of https://github.com/DSpace/dspace-angular/pull/3540
Complete feature documentation at https://github.com/kshepherd/dspace-geospatial-maps-doc/blob/main/README.md

## Description
Discovery and solr schema changes to support geospatial search filters and facets.

## Instructions for Reviewers
Apply this backend PR, ensure the new solr configuration is being read by solr and delete any existing search core data to force recreation with the updated schema. Perform a full reindex.

There is nothing specific to test in this PR, it just supports the frontend changes at https://github.com/DSpace/dspace-angular/pull/3540

## Checklist

- [x] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [x] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR **passes Checkstyle** validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR **includes Javadoc** for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR **passes all tests and includes new/updated Unit or Integration Tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.

kshepherd committed:
Discovery configuration to support geospatial search

EikLoe commented:
I tested the PR together with the angular-PR and it worked great! Thanks @kshepherd

toniprieto reviewed (approved):
Thanks! I just tested it, and the code looks good to me!

kshepherd committed:
Geospatial maps: Enable geospatial filters by default

Enabled in default search configuration.
NOT shown as facets by default.
Added to SearchFilterMatcher tests

kshepherd commented:
@tdonohue I have enabled the geospatial point and "has point" filters by default here, so that browse map click actions will work by default, since they filter on that point filter. The "has point / has coordinates" filter won't show in the facet list unless the user enables it though, as otherwise it would add a new facet to everybody's default search config even if they didn't want to use maps.

tdonohue reviewed (approved):
👍 Thanks @kshepherd !  My feedback here has also been addressed & I appreciate that you enabled the filters by default, as it allows for the various "enable" settings to work on the frontend with no changes to the backend required.
